### PR TITLE
Make push_downs work in table's read

### LINKED ISSUE #2632: Fuse engine seems ignore the projection column prune.
**Summary**

Description for this bug.

From @ZhiHanZ  wechat image.

image

```
>>> 3.79 * 1048576 * 1024 / 4284928
949.7199283068467  bytes/row
>>>
```

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

- Make push_downs work in table's read
- Add  `benefit_column_prune` in fuse engine which means it can benefit from projection columns
- Remove extra display of empty push_downs
- Remove  `get_single_node_table_io_context`

## Changelog

 
- Improvement
 

## Related Issues

Fixes #2632
Related  #2653

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/catalogs/table.rs [hunk 1]
```diff
@@ -117,6 +131,24 @@ pub trait ToReadDataSourcePlan {
...
+            table_info: table_info.clone(),
+            tbl_args: self.table_args(),
+
+            scan_fields: None,
+            parts: vec![],
+            statistics,
+            description,
+            push_downs: None,
+        })
+    }
```
Comment (drmingdrmer) [inline query/src/catalogs/table.rs:150]: Love it XDDD

Comment (sundy-li) [reply query/src/catalogs/table.rs]: It's removed... Hard to be compatible with lots tests cases..

**Comment by drmingdrmer** [comment]
Like a good teacher explaining a complex problem in a simple way. 👍

**Pull request merged**